### PR TITLE
Fix duplicate exceptions

### PR DESCRIPTION
**Pull request by heyams**

### CODE REVIEW

File: sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/LogDataMapper.java [hunk 1]
```diff
@@ -28,6 +30,8 @@ public class LogDataMapper {
 
     private static final ClientLogger logger = new ClientLogger(LogDataMapper.class);
 
+    private static final Set<String> EXCLUDE_LOGGER_NAMES = new HashSet<>(1);
```
Comment (trask) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/LogDataMapper.java:33]: probably worth initializing from a method to ensure [safe publication](https://shipilev.net/blog/2014/safe-public-construction/#_safe_publication), e.g.

```suggestion
    private static final Set<String> EXCLUDE_LOGGER_NAMES = getExcludeLoggerNames();
```

**Review by trask — APPROVED** [review]

**Review by jeanbisutti — APPROVED** [review]

File: sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/SpanDataMapper.java [hunk 2]
```diff
@@ -764,8 +765,10 @@ private void exportEvents(
                     // TODO (trask) map OpenTelemetry exception to Application Insights exception better
                     String stacktrace = event.getAttributes().get(SemanticAttributes.EXCEPTION_STACKTRACE);
                     if (stacktrace != null && !shouldSuppress.test(span, event)) {
-                        consumer.accept(
-                            createExceptionTelemetryItem(stacktrace, span, operationName, itemCount));
+                        Boolean exceptionAlreadyLogged = event.getAttributes().get(AiSemanticAttributes.EXCEPTION_LOGGED);
+                        if (exceptionAlreadyLogged == null || !exceptionAlreadyLogged) {
```
Comment (trask) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/SpanDataMapper.java:769]: I'm thinking `EXCEPTION_LOGGED` should be a String, so we can compare the stacktrace and only suppress if it's the same?

Comment (heyams) [reply sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/SpanDataMapper.java:769]: put the whole stacktrace into this attribute? not sure i understand your comment.

File: sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/AiSemanticAttributes.java [hunk 3]
```diff
@@ -97,6 +97,9 @@ public final class AiSemanticAttributes {
     public static final AttributeKey<Boolean> IS_PRE_AGGREGATED =
         AttributeKey.booleanKey("applicationinsights.internal.is_pre_aggregated");
 
+    public static final AttributeKey<String> LOGGED_EXCEPTION =
+        AttributeKey.stringKey("applicationinsights.internal.exception_logged");
```
Comment (trask) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/AiSemanticAttributes.java:101]: @heyams

**Pull request merged**